schema_enforcer/schemas/validator.py

```diff
@@ -0,0 +1,69 @@
+"""Classes for custom validator plugins."""
+# pylint: disable=no-member, too-few-public-methods
+import pkgutil
+import inspect
+from typing import Iterable
+import jmespath
+from schema_enforcer.validation import ValidationResult
+
+
+class ModelValidation:
```

Comment:
Looking at this class, I think it would make sense to implement a `BaseValidator` class that could be used here instead of `ModelValidation` but that could also be used by the other schema classes like `JsonSchema`. 

I wrote a quick class as an example. The main idea is to store the ValidationResults as part of the object and do not return them directly from `validate()`. We'll need to modify the SchemaManager to call `validate()` and `get_results()` but I think it can really simplify the `validate()` function. 
That wasn't as important to keep validate simple when we created the JsonSchema class initially but it can't hurt to simplify the code there as well.

```python
from schema_enforcer.validation import ValidationResult

class BaseValidation:
    """Base class for Validation classes."""

    def __init__(self):
       self._results = []

    def add_validation_error(self, message):
        self._results.append(ValidationResult(
            result="FAIL", 
            schema_id=self.id,
            message=message
        ))
        
    def get_results(self):
        """Return all validation results for this validator."""
        if not self._results:
            self._results.append(ValidationResult(
                result="PASS", 
                schema_id=self.id
            ))

        return self._results

    def validate(self, data: dict, strict: bool):
        """Required function for custom validator."""
        raise NotImplementedError
```

Thoughts ?
I can help to implement that if we agree this is the right direction

Reply:
I'm open to this approach since ModelValidation isn't providing much value.